tools/windows/install-help/cal/FinalizeInstall.cpp

```diff
@@ -133,21 +148,11 @@ UINT doFinalizeInstall(CustomActionData &data)
     }
     hr = 0;
 
-    if (!ddUserExists)
+    if (!AddUserToRequiredGroups(sid))
```

Comment:
This fails on read-only domain controllers. We want to keep the original `if` (so we only add users to groups when we created the user) but add an additional `or`  condition to always add the user to group when it's a local user.